Add documentation for ABS backup

feiskyer opened the pull request:
Closes #1987

/cc @fanminshi @mwieczorek

fanminshi commented on doc/user/abs_backup.md:
```diff
@@ -0,0 +1,95 @@
+
+- `storage-account`: the name of Azure storage account.
+- `storage-key`: the storage key of storage account.
+- `cloud`: the name of cloud environment, e.g. `AzurePublicCloud` or `AzureChinaCloud`.
```
If `cloud` is not specified, it is default to `AzurePublicCloud` right?

mwieczorek replied:
And should be marked as optional field, not required

feiskyer replied:
Let me address it as optional.

mwieczorek commented on doc/user/abs_backup.md:
````diff
@@ -0,0 +1,95 @@
+kubectl apply -f abs-secret.yaml
+```
+
+## Create EtcdBackup CRD
````
It’s just a naming, and I’m just nitpicking, but is it CRD (definition) or just an instance of CRD? So maybe just resource instead of CRD?

fanminshi replied:
Agree. it should be CR instead of CRD.

feiskyer replied:
yep, it should be CR

mwieczorek commented:
And general comment: do we want also EtcdRestore describe in the same doc?

feiskyer commented:
Addressed comments in the new commit.

@mwieczorek  @fanminshi While backup works properly, restore doesn't work in my cluster. Some foundings:

Etcd restore CR has changed its status to success

```
kubectl get etcdrestore example-etcd-cluster -o yaml 
apiVersion: etcd.database.coreos.com/v1beta2
kind: EtcdRestore
...
status:
  succeeded: true
```

And new pod has also been created

```
 kubectl get pod -l app=etcd,etcd_cluster=example-etcd-cluster
NAME                              READY     STATUS     RESTARTS   AGE
example-etcd-cluster-khv927lqq6   0/1       Init:0/3   0          6m36s
```

But the pod is always in PodInitializing state because `check-dns` init container failed to resolve dns 'example-etcd-cluster-khv927lqq6.example-etcd-cluster.etcd.svc':

```
nslookup: can't resolve 'example-etcd-cluster-khv927lqq6.example-etcd-cluster.etcd.svc'
nslookup: can't resolve 'example-etcd-cluster-khv927lqq6.example-etcd-cluster.etcd.svc'
Server:    10.0.0.10
Address 1: 10.0.0.10 kube-dns.kube-system.svc.cluster.local

Server:    10.0.0.10
Address 1: 10.0.0.10 kube-dns.kube-system.svc.cluster.local

nslookup: can't resolve 'example-etcd-cluster-khv927lqq6.example-etcd-cluster.etcd.svc'
Server:    10.0.0.10
Address 1: 10.0.0.10 kube-dns.kube-system.svc.cluster.local

nslookup: can't resolve 'example-etcd-cluster-khv927lqq6.example-etcd-cluster.etcd.svc'
```

@mwieczorek  @fanminshi Do you know why this is happening and how to fix this?

mwieczorek commented:
@feiskyer I haven't seen such issues. Which versions are you using (etcd-operator/etcd)?

@fanminshi I guess restore operator sets success status before etcd cluster gets healthy. Is it expected behaviour?

feiskyer commented:
@mwieczorek I'm using operators build from master branch. Seems headless service is removed by etcd-operator before restoring etcdcluster.

feiskyer commented:
@fanminshi Yep, it always fail in my testings. Should I add headless service creation first before creating pod `example-etcd-cluster-khv927lqq6`?

fanminshi commented:
@feiskyer 
`Seems headless service is removed by etcd-operator`
In restore operator, it [deletes the cluster CR](https://github.com/coreos/etcd-operator/blob/master/pkg/controller/restore-operator/sync.go#L131), and in return will delete any resources that's has the ownerRef of the cluster CR. I think that the headless svc has the ownerref of the cluster CR. Hence, headless service is also removed.

` Should I add headless service creation first before creating pod example-etcd-cluster-khv927lqq6` No, you don't need to do that. if you take a look of test for restore [testEtcdRestoreOperatorForS3Source](https://github.com/coreos/etcd-operator/blob/master/test/e2e/e2eslow/backup_restore_test.go#L170), the user doesn't need to create the headless service. The etcd-operator should have handled that.

I am currently investigating this issue. FYI. 

this pr looks good. The restore failure is another issue. Could you open it as a separate issue and so we can track it.

fanminshi commented:
@mwieczorek `I guess restore operator sets success status before etcd cluster gets healthy. Is it expected behaviour?`
The current code reports succeed if the seed pod is successfully created, However, it doesn't check pod is running or not. I would say the the behavior is not expected. The restore operator should only report succeed if the deployed pod is running. Could you file an issue to track that.

mwieczorek commented:
@fanminshi I created #1993 to track the Etcdrestore status issue.

fanminshi commented:
Thanks for the PR. merging!
